Unknown command-line option '--single-file'

[//]: # (
. Note: for support questions, please use Stackoverflow or Gitter**. 
. This repository's issues are reserved for feature requests and bug reports.
.
. In case of any problems with Allure Jenkins plugin** please use the following repository 
. to create an issue: https://github.com/jenkinsci/allure-plugin/issues
.
. Make sure you have a clear name for your issue. The name should start with a capital 
. letter and no dot is required in the end of the sentence. An example of good issue names:
.
. - The report is broken in IE11
. - Add an ability to disable default plugins
. - Support emoji in test descriptions
)

#### I'm submitting a ... 
  - [*] bug report
  - [ ] feature request
  - [ ] support request => Please do not submit support request here, see note at the top of this template.

#### What is the current behavior?
I am trying to use   '--single-file' in allureReport task 

i am running the command : ./gradlew allureReport --clean --single-file=true

I am getting error : 

FAILURE: Build failed with an exception.

* What went wrong:
Problem configuring task :functional-tests:allureReport from command line.
> Unknown command-line option '--single-file'.

The how to use documentation for all configuration options are missing in readme and allure documentation. 


#### If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem
Not sure 

#### What is the expected behavior?
It should generate report in a single html file. 

#### What is the motivation / use case for changing the behavior?


#### Please tell us about your environment:

| Allure version | id 'io.qameta.allure' version '2.12.0 |
| --- | --- |
| Test framework | Junit 5 |

| Generate report using | gradle allureReport task |

#### Other information 

[//]: # (
. e.g. detailed explanation, stacktraces, related issues, suggestions 
. how to fix, links for us to have more context, eg. Stackoverflow, Gitter etc
)

--- a/README.md
+++ b/README.md
@@ -345,6 +345,8 @@ allure {
 }
 ```
 
+To enable it for a single invocation without changing the DSL, run `./gradlew allureReport --single-file=true`.
+
 ### Running tests before building the report
 
 By default, `allureReport` task will NOT execute tests.

--- a/allure-report-plugin/src/main/kotlin/io/qameta/allure/gradle/report/tasks/AllureReport.kt
+++ b/allure-report-plugin/src/main/kotlin/io/qameta/allure/gradle/report/tasks/AllureReport.kt
@@ -39,6 +39,13 @@ abstract class AllureReport : AllureExecTask() {
         project.the<AllureExtension>().report.singleFile
     )
 
+    @Option(option = "single-file", description = "Generate a single-file Allure report")
+    fun setSingleFile(enabled: String) {
+        val parsed = enabled.toBooleanStrictOrNull()
+            ?: throw IllegalArgumentException("single-file expects true or false, got '$enabled'")
+        singleFile.set(parsed)
+    }
+
     @get:Internal
     val allure3ConfigFile = layout.file(
         providers.provider { temporaryDir.resolve("allurerc.json") }

--- a/allure-report-plugin/src/test/kotlin/io/qameta/allure/gradle/report/Allure3ReportIntegrationTest.kt
+++ b/allure-report-plugin/src/test/kotlin/io/qameta/allure/gradle/report/Allure3ReportIntegrationTest.kt
@@ -55,6 +55,26 @@ class Allure3ReportIntegrationTest {
     }
 
     @Test
+    fun `allureReport should accept single-file as a task option`() {
+        assumeFalse("Fake Allure 3 runtime tests currently support Unix-like systems only", Os.isFamily(Os.FAMILY_WINDOWS))
+
+        val projectDir = createAllure3Project(singleFile = false)
+
+        val buildResult = runner(projectDir)
+            .withArguments(commonArgs("allureReport", "--single-file=true"))
+            .build()
+
+        assertThat(buildResult.task(":allureReport")?.outcome)
+            .isEqualTo(TaskOutcome.SUCCESS)
+
+        val configFile = projectDir.resolve("build/tmp/allureReport/allurerc.json")
+        assertThat(configFile)
+            .exists()
+        assertThat(configFile.readText())
+            .contains("\"singleFile\": true")
+    }
+
+    @Test
     fun `allureServe should run open for Allure 3`() {
         assumeFalse("Fake Allure 3 runtime tests currently support Unix-like systems only", Os.isFamily(Os.FAMILY_WINDOWS))
 

